app/scripts/lib/errors.js

```diff
@@ -253,8 +253,10 @@ define(function (require, exports, module) {
     normalizeXHRError: function (xhr) {
       var err;
 
-      if (! xhr || xhr.status === 0) {
+      if (! xhr || xhr.status === 0 || xhr.status === 503 ) {
```

Comment:
Normalization in this context is to convert an error response format from an external server to a format we use internally on the front end. In this context, our backend server can return a 503 or 429 http error response that contains xhr.responseJSON. We do not want to change the existing behavior for these responses because they are already use a format the frontend can work with. The backend server can also respond with a 503 or 429 http error response that does *not* contain xhr.responseJSON. This can happen if we have put the backend server into maintenance mode. This is the type of error we want to convert because the error is not understood by the front end.

Don't be afraid to add one or more branches to the `if/else-if` logic, it may work out to be cleaner that way, e.g.:

```js
if (! xhr || xhyr.status === 0) {
  err = this.toError('SERVICE_UNAVAILABLE');
} else if (xhr.responseJSON) {
  err = this.toError(xhr.responseJSON)
} else if (some check here) {
  err = this.toError(something here);
} else {
  err = this.toError('UNEXPECTED_ERROR');
}
```
Going through this a bit more, when you convert 429 errors, instead of using `Too Many Requests`, can you simply use `THROTTLED`? `THROTTLED` is the error we use internally and one we already have logs set up for.